addons/project/models/project_task.py

```diff
@@ -979,7 +979,8 @@ def write(self, vals):
                     task.recurrence_id = recurrence.id
 
         if 'recurring_task' in vals and not vals.get('recurring_task'):
-            self.recurrence_id.unlink()
+            if not self.env['project.task'].search_count([('recurrence_id', 'in', self.recurrence_id.ids)]) > 1:
+                self.recurrence_id.unlink()
```

Comment:
Sounds a little too easy :smile: 
Have you tried changing the state of the last task of the recurrence to done?
I guess the idea is good (saying you can have "recurring_task" false, but still keep the "recurrence_id"),
but now that we're differentiating those 2 concepts, we need to ensure that eachone is properly used elsewhere :angel:

Reply:
It would be useful to write some tests here :pray:

Reply:
As you suggest, changes are done and also write test cases for that 
Thanks